internal/util/importutilv2/parquet/util.go

```diff
@@ -105,77 +93,164 @@ func CreateFieldReaders(ctx context.Context, fileReader *pqarrow.FileReader, sch
 	return crs, nil
 }
 
-func convertArrowSchemaToDataType(field arrow.Field, isList bool) (schemapb.DataType, bool) {
-	switch field.Type.ID() {
+func isArrowIntegerType(dataType arrow.Type) bool {
+	switch dataType {
+	case arrow.INT8, arrow.INT16, arrow.INT32, arrow.INT64:
+		return true
+	default:
+		return false
+	}
+}
+
+func isArrowFloatingType(dataType arrow.Type) bool {
+	switch dataType {
+	case arrow.FLOAT32, arrow.FLOAT64:
+		return true
+	default:
+		return false
+	}
+}
+
+func isArrowArithmeticType(dataType arrow.Type) bool {
+	return isArrowIntegerType(dataType) || isArrowFloatingType(dataType)
+}
+
+func isArrowDataTypeConvertible(src arrow.DataType, dst arrow.DataType) bool {
+	srcType := src.ID()
+	dstType := dst.ID()
+	switch srcType {
 	case arrow.BOOL:
-		return schemapb.DataType_Bool, false
+		return dstType == arrow.BOOL
 	case arrow.UINT8:
-		if isList {
-			return schemapb.DataType_BinaryVector, false
-		}
-		return schemapb.DataType_None, false
+		return dstType == arrow.UINT8
 	case arrow.INT8:
-		return schemapb.DataType_Int8, false
+		return isArrowArithmeticType(dstType)
 	case arrow.INT16:
-		return schemapb.DataType_Int16, false
+		return isArrowArithmeticType(dstType) && dstType != arrow.INT8
 	case arrow.INT32:
-		return schemapb.DataType_Int32, false
+		return isArrowArithmeticType(dstType) && dstType != arrow.INT8 && dstType != arrow.INT16
```

Comment:
Correct me if I'm wrong, srcType is from the input file, dstType is from milvus schema, so when srcType is arrow.Int32, dstType can be `INT8` or `INT16` or `INT32`.